app/models/dbedt_upload.rb

```diff
@@ -42,6 +52,22 @@ def make_active
     self.update! active: true
   end
 
+  def get_status(which)
+    if which == 'cats'
+      self.cats_status
+    else
```

Comment:
we could see if self[which + '_status'].nil? and then return that status if it exists.